### PR TITLE
New chan type: 'httpserver'

### PR DESCRIPTION
**Pull request by jsccast**

### CODE REVIEW

**Comment by jsccast** [comment]
Next task: Write something to generate decent documentation for all channel types.

File: chans/httpserver.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+	return c.To(ctx, m)
+}
+
+func (c *HTTPServer) Recv(ctx *dsl.Ctx) chan dsl.Msg {
+	return c.reqs
+}
+
+func (c *HTTPServer) Kill(ctx *dsl.Ctx) error {
+	return dsl.Brokenf("%T doesn't support 'Kill'", c)
+}
```
Comment (carlos-loya) [inline chans/httpserver.go:199]: I think supporting a `Kill` step for the httpserver channel could lead to some interesting chaos plax tests

**Review by carlos-loya — APPROVED** [review]
There could be potential for injecting chaos into the http server channel ala [Toxiproxy](https://github.com/Shopify/toxiproxy). I think it should probably be a separate channel, `badhttpserver`?

The code looks good and I like the use of the pattern field for the handler configuration.

**Review by robertfarnum — APPROVED** [review]
Nice addition

**Pull request merged**